boa/src/builtins/array/mod.rs

```diff
@@ -122,11 +113,36 @@ pub fn make_array(this: &Value, args: &[Value], ctx: &mut Interpreter) -> Result
     // to its Javascript Identifier (global constructor method name)
     this.set_kind(ObjectKind::Array);
 
-    // And finally add our arguments in
-    for (n, value) in args.iter().enumerate() {
-        this.set_field_slice(&n.to_string(), value.clone());
+    // add our arguments in
+    let mut length = args.len() as i32;
+    match args.len() {
+        1 if args[0].is_integer() => {
+            length = from_value::<i32>(args[0].clone()).expect("Could not convert argument to i32");
+            // TODO: It should not create an array of undefinds, but an empty array ("holy" array in V8) with length `n`.
+            for n in 0..length {
+                this.set_field_slice(&n.to_string(), Gc::new(ValueData::Undefined));
+            }
+        }
+        1 if args[0].is_double() => {
+            // TODO: throw `RangeError`.
+            unimplemented!("RangeError: invalid array length");
```

Comment:
I would just use `panic!()` for now, since it's not that the feature is unimplemented, it's just that we cannot throw errors.